### ISSUE TITLE
[Glide][GIFs] GIFs randomly glitching/stopping/flickering

### ISSUE THREAD

**Issue by drinkthestars** [google/accompanist#241]
## Describe the bug
When loading GIFs into `GlideImage`s, some GIFs seem to randomly glitch/stop playing/flicker. These `GlideImage`s are being loaded inside of a `LazyColumn`.

`GlideImage` is being used in the following manner:
``` Kotlin
    GlideImage(
        modifier = Modifier.fillMaxSize(),
        requestBuilder = { apply(RequestOptions().centerCrop()) },
        data = "GIF URL",
        contentDescription = null,
        contentScale = ContentScale.Crop
    )
```

The full code can be found [here](https://github.com/drinkthestars/compose-lab/blob/main/app/src/main/kotlin/com/goofy/goober/composelab/gifs/GifResults.kt).

## To Reproduce
Load some GIFs from network and wait, scroll around.

## Expected behavior
GIFs should play normally

## Screenshots?
Videos of the issue:

https://user-images.githubusercontent.com/2978958/111103849-96bcf600-850c-11eb-9abd-8c2ce4e73986.mp4

https://user-images.githubusercontent.com/2978958/111103844-93c20580-850c-11eb-8325-4d2fb8a2d2fb.mp4

## Environment:
 - Android OS version: 11
 - Device: Pixel 3a XL
 - Accompanist version: `0.6.2`

**Comment by chrisbanes** [comment]
Thanks for the report. This has been reported before on Slack by @davols, and I investigated briefly. I couldn't find anything in Accompanist which would trigger this, so there's something weird going on.

If you use `CoilImage` and it's [GIF support](https://google.github.io/accompanist/coil/#gifs), it works perfectly. They're both based on the same integration code ([`AndroidDrawablePainter`](https://google.github.io/accompanist/api/imageloading-core/imageloading-core/dev.chrisbanes.accompanist.imageloading/-android-drawable-painter/)) so this is probably something in the way which Glide decodes GIF frames. I haven't had a chance to take a deeper look yet.

**Comment by davols** [comment]
fwiw:
I have seen an issue where you add an item to `LazyColumn` and it seems to be using the wrong url for the image even with Coil. That said, Coil works perfectly otherwise. Will try to see if I can investigate and make a sample app with that issue later this week.

**Comment by chrisbanes** [comment]
Interesting, that sounds like a different issue! Thanks

**Comment by drinkthestars** [comment]
> If you use CoilImage and it's GIF support, it works perfectly. They're both based on the same integration code (AndroidDrawablePainter) so this is probably something in the way which Glide decodes GIF frames. I haven't had a chance to take a deeper look yet.

Gotcha, thank you!

> I have seen an issue where you add an item to LazyColumn and it seems to be using the wrong url for the image,as in wrong gif in wrong item, even with Coil. That said, Coil works perfectly otherwise. Will try to see if I can investigate and make a sample app with that issue later this week.

Right, with `GlideImage` I also notice that the wrong gifs get bound to the wrong items. Sometimes, after scrolling away and coming back, the right item seems to get bound.

I had to use Glide because I'm trying to create a Compose equivalent of a screen that's written using Android Views, so didn't want to include an additional Coil dependency.

I'll try using `Coil` to see how it goes, though, and whether or not I see the issue @davols mentioned above.

**Comment by drinkthestars** [comment]
Tried this out with `coil`, and the integrity of items + positions seems to be preserved, plus no glitching. However, I do see that when the items return to view, they reload with the fade-in animation even though they were previously "bound" (does use `memoryCachePolicy(CachePolicy.ENABLED)`). 

Here's both `coil` & `glide` treatments for comparison:


| `coil` | `glide`
-- | --
<img src="https://user-images.githubusercontent.com/2978958/113550082-57cc0e80-95a7-11eb-9f0e-ec70ed45ff90.gif" width=250/> | <img src="https://user-images.githubusercontent.com/2978958/113550403-db85fb00-95a7-11eb-990f-46dfc0baf961.gif" width=250/>



 


---



**Versions:**
Compose: `beta03` 
Accompanist: `0.7.0`

**Comment by ashar-7** [comment]
I have sizing problems with GIFs. The GIFs ignore the `fillMaxSize` modifier and take up a size of their own while this is not the case with normal images. The problem exists in both Coil and Glide.


| Normal | GIF |
| -------- |  --- |
| <img src=https://user-images.githubusercontent.com/26629623/113599411-36792a00-965c-11eb-8b1a-5beb0e00a02a.png width=220/> | <img src=https://user-images.githubusercontent.com/26629623/113599418-3842ed80-965c-11eb-93a1-c0100b998c05.png width=220/> |

**Comment by chrisbanes** [comment]
> I have sizing problems with GIFs. The GIFs ignore the `fillMaxSize` modifier and take up a size of their own while this is not the case with normal images. The problem exists in both Coil and Glide.

I'm guessing that you're not setting an appropriate `contentScale`. I assume that you want `ContentScale.Crop`.

**Comment by chrisbanes** [comment]
> Tried this out with `coil`, and the integrity of items + positions seems to be preserved, plus no glitching. However, I do see that when the items return to view, they reload with the fade-in animation even though they were previously "bound" (does use `memoryCachePolicy(CachePolicy.ENABLED)`).

Please raise a new issue. This issue is specifically for the glitching behavior seen by using Glide.

**Comment by ashar-7** [comment]
> > I have sizing problems with GIFs. The GIFs ignore the `fillMaxSize` modifier and take up a size of their own while this is not the case with normal images. The problem exists in both Coil and Glide.
> 
> I'm guessing that you're not setting an appropriate `contentScale`. I assume that you want `ContentScale.Crop`.

I actually need `ContentScale.FillWidth`. I tried out with `ContentScale.Crop` and it does seem to work for the GIFs but the other images get cropped out which I don't want.

**Comment by chrisbanes** [comment]
> I actually need `ContentScale.Fit`. I tried out with `ContentScale.Crop` and it does seem to work for the GIFs but the other images get cropped out which I don't want.

Please raise a new issue. I think this is actually fixed in the big refactor in #220.

**Comment by chrisbanes** [comment]
Just spent 2 hours trying to dig into why this is happening and come up empty. My guess is that Glide's `GifDrawable` has a lot of timing assumptions in it, which don't work very well with how Compose draws.

Considering that we have a valid working workaroud (== use Coil instead), I'm ice-boxing this issue.
